symengine/derivative.cpp

```diff
@@ -80,6 +80,25 @@ class DiffImplementation
                    self.get_a()->diff(x));
     }
 
+    static RCP<const Basic> diff(const Dirichlet_eta &self,
+                                 const RCP<const Symbol> &x)
+    {
+        RCP<const Basic> m1;
+        m1 = zeta(self.get_arg())->diff(x);
+        // check https://en.wikipedia.org/wiki/Dirichlet_eta_function
+        // for the equation
+        if (eq(*self.get_arg(), *one)) {
```

Comment:
This doesn't make sense. `dirichlet_eta(one)->diff(x)` is zero, no matter what the symbol `x` is.